### ISSUE TITLE
Destructuring Defaults Spec

### ISSUE THREAD

**Issue by xjamundx** [estree/estree#13]
Basically an issue to make sure the latest discussion from this thread is included in the docs:
https://bugzilla.mozilla.org/show_bug.cgi?id=932080

**Comment by RReverser** [comment]
I'll just put here some details from discussion results:

* In `[a = 1]`, `{x = 2}`, `{x: x = 2}` we parse the default value as:

```typescript
interface AssignmentPattern <: Pattern {
  left: Pattern;
  right: Expression;
}
```

so as in any other patterns, we resemble structure of corresponding expression type.

* In `[...x]` we parse the rest part as

```typescript
interface RestElement <: Pattern {
  argument: Identifier | ArrayPattern;
}
```

* For function params we parse default parameters and rest parameter just in the same way as if they were elements of `ArrayPattern` for consistency instead of introducing new properties on function node (see #3).

**Comment by xjamundx** [comment]
Thanks @RReverser for the clarity

**Comment by RReverser** [comment]
Closed by #37.
